### PR TITLE
add task solution

### PR DESCRIPTION
**Pull request by bashtovaA**
- [ДЕМО ПОСИЛАННЯ](https:// bashtovaA .github.io/layout_product-cards/)
- [ПОСИЛАННЯ НА ЗВІТ ПРО ВИПРОБУВАННЯ](https:// bashtovaA .github.io/layout_product-cards/report/html_report/)

### CODE REVIEW

**Comment by bashtovaA** [comment]
> * [ДЕМО ПОСИЛАННЯ](https:// bashtovaA .github.io/layout_product-cards/)
> * [ПОСИЛАННЯ НА ЗВІТ ПРО ВИПРОБУВАННЯ](https:// bashtovaA .github.io/layout_product-cards/report/html_report/)

File: src/index.html [hunk 1]
```diff
@@ -5,9 +5,41 @@
     <meta name="viewport"
           content="width=device-width, user-scalable=no, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0">
     <title>Product cards</title>
-    <link rel="stylesheet" href="./style.css">
+    <link rel="stylesheet" href="./styles/index.scss">
   </head>
   <body>
-    <h1>Product cards</h1>
+    <div class="container" data-qa="card">
+      <img class="container__images"
```
Comment (mykhalenych) [inline src/index.html:12]: Screenshot 2023-07-12 at 16 53 30

File: src/index.html [hunk 2]
```diff
@@ -5,9 +5,41 @@
...
+      src="images/imac.jpeg"
+      alt="APPLE A1419 iMac 27">
+
+      <h1 class="container__name">APPLE A1419 iMac 27" Retina 5K Monoblock (MNED2UA/A)</h1>
```
Comment (mykhalenych) [inline src/index.html:16]: change to h2

File: src/styles/blocks/card.scss [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+  font-family: "Roboto", Helvetica, sans-serif;
+  font-style: normal;
+}
+
+body {
+  margin: 0;
+}
+
+.container {
+  float: left;
```
Comment (mykhalenych) [inline src/styles/blocks/card.scss:16]: why do you need this?

File: src/styles/blocks/card.scss [hunk 4]
```diff
@@ -0,0 +1,132 @@
+$color1: #fff;
```
Comment (mykhalenych) [inline src/styles/blocks/card.scss:1]: create better naming instead 1, 2, 3, 4

File: src/styles/blocks/card.scss [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+
+.container {
+  float: left;
+  width: 200px;
+  border-radius: 5px;
+  background-color: #{$color1};
+  border: 1px solid #f3f3f3;
+  display: flex;
+  flex-direction: column;
+  box-sizing: border-box;
```
Comment (mykhalenych) [inline src/styles/blocks/card.scss:23]: move to `*` selector

**Review by mykhalenych — CHANGES_REQUESTED** [review]

File: src/styles/blocks/card.scss [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+  &__name {
+    font-size: 12px;
+    font-weight: 500;
+    line-height: 18px;
+    color: #{$color-b};
+    display: flex;
+    margin: 0;
+    width: 166px;
+    word-break: break-word;
+    margin-bottom: 4px;
```
Comment (Viktor-Kost) [inline src/styles/blocks/card.scss:36]:
```suggestion
```

File: src/styles/blocks/card.scss [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+    margin-top: 40px;
+  }
+
+  &__code {
+    font-size: 10px;
+    font-weight: 400;
+    line-height: 14px;
+    color: #{$cont-color};
+    display: flex;
+    margin: 0;
```
Comment (Viktor-Kost) [inline src/styles/blocks/card.scss:46]:
```suggestion
    margin-top: 4px;
```

File: src/styles/blocks/card.scss [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+$color-b: #060b35;
+$cont-color: #616070;
+$blue-cont: #00acdc;
+
+html {
+  font-family: "Roboto", Helvetica, sans-serif;
+  font-style: normal;
+}
+
+body {
```
Comment (Viktor-Kost) [inline src/styles/blocks/card.scss:11]: reset default margin for p and h2 here
```suggestion
body,
p,
h2 {
```

File: src/styles/blocks/card.scss [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+  padding-top: 32px;
+  padding-bottom: 16px;
+
+  &__name {
+    font-size: 12px;
+    font-weight: 500;
+    line-height: 18px;
+    color: #{$color-b};
+    display: flex;
+    margin: 0;
```
Comment (Viktor-Kost) [inline src/styles/blocks/card.scss:33]:
```suggestion
```

File: src/styles/blocks/card.scss [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+    width: 166px;
```
Comment (Viktor-Kost) [inline src/styles/blocks/card.scss:34]: no needs to hardcode width
```suggestion
```

File: src/styles/blocks/card.scss [hunk 11]
```diff
@@ -0,0 +1,131 @@
...
+    color: #{$cont-color};
+    display: flex;
+    margin: 0;
+  }
+
+  &__images {
+    display: flex;
+    height: 134px;
+    padding-right: 3px;
+    padding-left: 3px;
```
Comment (Viktor-Kost) [inline src/styles/blocks/card.scss:53]:
```suggestion
    margin: 0 auto;
```

File: src/styles/blocks/card.scss [hunk 12]
```diff
@@ -0,0 +1,131 @@
...
+    line-height: 16px;
+    text-transform: uppercase;
+    color: #{$w-cont};
+    border-radius: 5px;
+    background-color: #{$blue-cont};
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    margin-top: 16px;
+    height: 40px;
```
Comment (Viktor-Kost) [inline src/styles/blocks/card.scss:105]: it's not a good practice to hardcode height for element with content inside. Just use padding: 11px 0;
```suggestion
```

File: src/styles/blocks/card.scss [hunk 13]
```diff
@@ -0,0 +1,131 @@
...
+    background: center no-repeat url(../images/star-active.svg);
+    width: 16px;
+    height: 16px;
+    margin-right: 4px;
+    padding: 0;
+    display: flex;
+  }
+}
+
+.stars :nth-child(5) {
```
Comment (Viktor-Kost) [inline src/styles/blocks/card.scss:128]: add solution from previous task to make amount of stars be flexible

File: src/styles/blocks/card.scss [hunk 14]
```diff
@@ -0,0 +1,131 @@
...
+}
+
+.stars {
+  display: flex;
+
+  &__star {
+    background: center no-repeat url(../images/star-active.svg);
+    width: 16px;
+    height: 16px;
+    margin-right: 4px;
```
Comment (Viktor-Kost) [inline src/styles/blocks/card.scss:122]: all stars besides last one should have margin

File: src/styles/blocks/card.scss [hunk 15]
```diff
@@ -0,0 +1,131 @@
...
+html {
+  font-family: "Roboto", Helvetica, sans-serif;
+  font-style: normal;
+}
+
+body {
+  margin: 0;
+}
+
+.container {
```
Comment (Viktor-Kost) [inline src/styles/blocks/card.scss:15]: I'd recommend to name your card just .card

File: src/styles/blocks/card.scss [hunk 16]
```diff
@@ -0,0 +1,131 @@
...
+  width: 200px;
+  border-radius: 5px;
+  background-color: #{$w-cont};
+  border: 1px solid #f3f3f3;
+  display: flex;
+  flex-direction: column;
+  box-sizing: border-box;
+  padding: 0 16px;
+  padding-top: 32px;
+  padding-bottom: 16px;
```
Comment (Viktor-Kost) [inline src/styles/blocks/card.scss:25]: use padding shorthand
```suggestion
  padding: 32px 16px 16px;
```

File: src/index.html [hunk 17]
```diff
@@ -5,9 +5,43 @@
...
+          <div class="stars__star"></div>
+          <div class="stars__star"></div>
+          <div class="stars__star"></div>
+        </div>
+        <p class="container__reviews">Reviews: 5</p>
+      </div>
+
+      <div class="container__price">
+        <div class="container__text">Price:</div>
+        <div class="container__value">$2,199</div>
```
Comment (Viktor-Kost) [inline src/index.html:34]: use semantic tags for elements with text inside (`<p>`)

File: src/styles/blocks/card.scss [hunk 18]
```diff
@@ -0,0 +1,131 @@
+$w-cont: #fff;
+$color-b: #060b35;
+$cont-color: #616070;
+$blue-cont: #00acdc;
```
Comment (Viktor-Kost) [inline src/styles/blocks/card.scss:4]:
```suggestion
$white-color-primary: #fff;
$black-color-primary: #060b35;
$neutral-color-primary: #616070;
$accent-color-primary: #00acdc;
```

**Review by Viktor-Kost — CHANGES_REQUESTED** [review]
Hello!
I see not all previous remarks were fixed.
As an addition to my comments pay attention to these unresolved conversations
https://github.com/mate-academy/layout_product-cards/pull/3316#discussion_r1261217558
https://github.com/mate-academy/layout_product-cards/pull/3316#discussion_r1261219606

If you need any help - feel free to ask at fe_chat

File: src/styles/blocks/card.scss [hunk 19]
```diff
@@ -0,0 +1,135 @@
...
+    margin-right: 4px;
+    padding: 0;
+    display: flex;
+  }
+}
+
+.stars--2 :nth-child(-n+1),
+.stars--2 :nth-child(-n+2),
+.stars--2 :nth-child(-n+3),
+.stars--2 :nth-child(-n+4) {
```
Comment (Viktor-Kost) [inline src/styles/blocks/card.scss:129]: suppose it should looks like
```suggestion
.stars--1 :nth-child(-n+1),
.stars--2 :nth-child(-n+2),
.stars--3 :nth-child(-n+3),
.stars--4 :nth-child(-n+4),
.stars--5 :nth-child(-n+5) {
```

File: src/styles/blocks/card.scss [hunk 20]
```diff
@@ -0,0 +1,135 @@
...
+  background-image: url(../images/star-active.svg);
+}
+
+.stars--2 :last-child {
+  margin-right: 0;
+}
```
Comment (Viktor-Kost) [inline src/styles/blocks/card.scss:135]:
```suggestion
```

File: src/styles/blocks/card.scss [hunk 21]
```diff
@@ -0,0 +1,135 @@
...
+
+.stars {
+  display: flex;
+
+  &__star {
+    background: center no-repeat url(../images/star.svg);
+    width: 16px;
+    height: 16px;
+    margin-right: 4px;
```
Comment (Viktor-Kost) [inline src/styles/blocks/card.scss:120]:
```suggestion
```

File: src/styles/blocks/card.scss [hunk 22]
```diff
@@ -0,0 +1,135 @@
...
+    padding: 0;
+    display: flex;
```
Comment (Viktor-Kost) [inline src/styles/blocks/card.scss:122]:
```suggestion
    display: flex;
    
    &:not(:last-child) {
      margin-right: 4px;
    }
```

**Review by Viktor-Kost — CHANGES_REQUESTED** [review]
Hi!)

I see this moment is still not fixed
https://github.com/mate-academy/layout_product-cards/pull/3316#discussion_r1262515243
Please take a look at code styling recommendations from a [checklist](https://github.com/mate-academy/layout_product-cards/blob/master/checklist.md)

If you need any help - feel free to ask at fe_chat

**Review by etojeDenys — CHANGES_REQUESTED** [review]
image
image
fix this tag

**Review by etojeDenys — CHANGES_REQUESTED** [review]

**Review by VitaliyBondarenko1982 — APPROVED** [review]

**Pull request closed without merging**